app/models/registry.rb

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env ruby
+
+
+class Registry
+  class NoBearerRealmException < RuntimeError
+  end
+
+  class AuthorizationError < RuntimeError
+  end
+
+  class ManifestNotFoundError < RuntimeError
+  end
+
+  class CredentialsMissingError < RuntimeError
+  end
+
+  def initialize(host, use_ssl=true, username=nil, password=nil)
+    @host = host
+    @use_ssl = use_ssl
+    @base_url = "http#{'s' if @use_ssl}://#{@host}/v2/"
+    @username = username
+    @password = password
+  end
+
+  def has_credentials?
+    @username && @password
+  end
+
+  def manifest(repository, tag='latest')
+    res = get_request("#{repository}/manifests/#{tag}")
+    if res.code.to_i == 200
+      JSON.parse(res.body)
+    elsif res.code.to_i == 404
+      fail ManifestNotFoundError, "Cannot find manifest for #{repository}:#{tag}"
+    else
+      fail "Something went wrong while fetching manifest for #{repository}:#{tag}:" \
+        "[#{res.code}] - #{res.body}"
+    end
+  end
+
+  def get_request(path, request_auth_token=true)
+    uri = URI.join(@base_url, path)
+    req = Net::HTTP::Get.new(uri)
```

Comment:
You can sit on top of RestClient for ruby which gives you all the goodies out of the box